query_object.rb

```diff
@@ -1,45 +1,58 @@
-module YourSuperModule
+module ParamsMethods
   def self.included(host_class)
     host_class.extend(ClassMethods)
   end
 
   def initialize(params)
-    params.each do |k, v|
-      next eval("set_#{k}('#{v}')") if v.instance_of?(String)
-      eval("set_#{k}(#{v})")
-    end
+    @params = params
+  end
+
+  def query
+    params_for_query = available_params.map do |p|
+      "#{p}: #{value_for(p)}" if value_for(p)
+    end.compact
+    "Select with: #{params_for_query}"
+  end
+
+  def params
+    @params
+  end
+
+  def available_params
+    self.class.available_params
+  end
+
+  def default_value
```

Comment:
Why do you use singular form here? Don't we have multiple default values in this collection?

Reply:
Fixed